text/0400-appmonitor-l2.md

```diff
@@ -0,0 +1,499 @@
+with just a few lines of code, enabling you to retrieve code snippets outside of the management console,
+which was previously impossible. This allows users to retrieve code snippets
+and fully automate deployments without having to go through the management console.
+
+## Working Backwards
+
+### CHANGELOG
+
+`feat(rum): AppMonitor L2; Import, Generate code snippet`
+
+### README
+
+---
+
+# Amazon CloudWatch RUM Construct Library
+
+## Table of Contents
+
+- [Introduction](#introduction)
+  - [App Monitor](#appmonitor)
+  - [Authorizer](#authorizer)
+    - [Existing Cognito ID pool](#use-an-existing-amazon-cognito-identity-pool)
+    - [Third-party provider](#use-third-party-provider)
+  - [Code Snippet](#code-snippet)
+    - [RUM web client configuration](#rum-web-client-configuration)
+    - [With s3 deployment](#with-s3-deployment)
+  - [Importing App Monigor](#importing-app-monitor)
+
+## Introduction
+
+CloudWatch RUM is monitoring tool that can perform real user monitoring to collect
+and view client-side data about your web application performance from actual user sessions in near real time.
+
+This module supports the ability for users to create CloudWatch RUM and retrieve code snippets on CloudFormation.
+
+## App Monitor
+
+Define `AppMonitor` to your stack:
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor'
+});
+```
+
+### Authorizer
+
+By default, AppMonitor creates a new Amazon Cognito identity pool and uses it to authenticate the put event.
+
+#### Use an existing Amazon Cognito identity pool
+
+If you want to use an existing Amazon Cognito identity pool, you need to pass a `CognitoIdentityPoolAuthorizer` instance.
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+import * as iam from '@aws-cdk/aws-iam';
+
+declare const myRole: iam.IRole;
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor',
+  authorizer: new rum.CognitoIdentityPoolAuthorizer({
+    identityPoolId: 'my-user-pool-id',
+    unauthenticatedRole: myRole,
+  }),
+});
+```
+
+#### Use Third-party provider
+
+You can also use a third party authenticator as in `CognitoIdentityPoolAuthorizer`.
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+import * as iam from '@aws-cdk/aws-iam';
+
+declare const myRole: iam.IRole;
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor',
+  authorizer: new rum.ThirdPartyAuthorizer({
+    role: myRole,
+  }),
+});
+```
+
+### Code Snippet
+
+AppMonitor generates a code snippet that looks like to create on the management console. Note, however,
+that unlike the management console, the code snippets do not have `<script>` tags because the CDK expects them to be automatically embedded in the application.
```

Comment:
Can you explain further why the generated code snippet should not include the `<script>` tag?